File: Ereceipt/Ereceipt.Domain/Models/Comment.cs
using System;
using System.ComponentModel.DataAnnotations;
namespace Ereceipt.Domain.Models
{
    public class Comment : BaseModelWithIdentityGen<long>
    {
        [Required, MinLength(1), MaxLength(500)]
        public string Text { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public Guid ReceiptId { get; set; }
        public Receipt Receipt { get; set; }
    }
}

File: Ereceipt/Ereceipt.Web/ClientApp/src/app/comment/comment.ts
import { User } from './user/user';
import { Receipt } from './receipt/receipt';
export class Comment {
    constructor(
        public id: number,
        public createdAt: Date,
        public text: string,
        public user: User,
        public receipt: Receipt,
        public updatedAt: Date
    ) { }
}